valhalla/jawn/src/lib/stores/LogStore.ts

```diff
@@ -381,4 +378,38 @@ export class LogStore {
 
     return Array.from(entryMap.values());
   }
+
+  filterDuplicateSearchRecords(
+    entries: Database["public"]["Tables"]["request_response_search"]["Insert"][]
+  ) {
+    const entryMap = new Map<
+      string,
+      Database["public"]["Tables"]["request_response_search"]["Insert"]
+    >();
+
+    entries.forEach((entry) => {
+      if (!entry.request_id) {
+        return;
+      }
+
+      const existingEntry = entryMap.get(entry.request_id);
+
+      // No existing entry, add it
+      if (!existingEntry || !existingEntry.created_at) {
+        entryMap.set(entry.request_id, entry);
+        return;
+      }
+
+      if (
+        (entry.created_at &&
+          new Date(entry.created_at) < new Date(existingEntry.created_at)) ||
+        !existingEntry.request_body_vector ||
```

Comment:
We probably want to also check the new entry. If the request_body_vector and response_body_vector actually exists.